### PR TITLE
fix minor mistake

### PR DESCRIPTION
**Pull request by Tw1stFate**
Use the `flatMap` method. Parameter transform: A closure that takes the unwrapped value, right?
and the example's result is 1764, not 1746.

### CODE REVIEW

File: stdlib/public/core/Optional.swift [hunk 1]
```diff
@@ -182,15 +182,15 @@ public enum Optional<Wrapped> : ExpressibleByNilLiteral {
...
+  ///     // Prints "Optional(1764)"
   ///
   /// - Parameter transform: A closure that takes the unwrapped value
   ///   of the instance.  
   /// - Returns: The result of the given closure. If this instance is `nil`,
   ///   returns `nil`.
   @_inlineable
   public func flatMap<U>(
-    _ transform: (Wrapped) throws -> U?
+    _ transform: (Unwrapped) throws -> U?
```
Comment (LuizZak) [inline stdlib/public/core/Optional.swift]: Will this compile? I assume `Wrapped` is the type that comes from the generic parametrization of `Optional<Wrapped>`, i.e. it's not really a parameter name that can be changed.

Comment (CodaFi) [inline stdlib/public/core/Optional.swift]: This change is also outside the scope of the documentation change and should be reverted.

Comment (RushingTwist) [inline stdlib/public/core/Optional.swift]: sorry, my English is pool, and can I think 'Wrapped' is the value  unwrapped from a wrapped value ?

**Comment by CodaFi** [comment]
Good catch, thanks!

⛵️

**Comment by CodaFi** [comment]
@swift-ci please smoke test
